File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-1.java
class Solution {
    private int[] memo ;
    public int minCostClimbingStairs(int[] cost) {
        if (cost.length<=2){
            return Math.min(cost[0],cost[1]);
        }
        memo = new int[cost.length];
        memo[0]=cost[0];
        memo[1]=cost[1];
        for (int i = 2;i<cost.length;i++){
            memo[i]=+Math.min(cost[i-1]+memo[i-1],cost[i-2]+memo[i-2]);
            System.out.println(memo[i]);
        }
        
        return memo[cost.length-1];
        
    }

}
/**
memo=[1,2,]
*/

File: Data Structures & Algorithms/minimum-changes-to-make-alternating-binary-string/submission-1.py
class Solution:
    def minOperations(self, s: str) -> int:
        w1=0
        w2=0
        for i in range(len(s)) :
            if i%2==0 :
                if s[i]=='0' :
                    w1+=1
            else :
                if s[i]=='1':
                    w1+=1
        
        for i in range(len(s)) :
            if i%2==0 :
                if s[i]=='1' :
                    w2+=1
            else :
                if s[i]=='0':
                    w2+=1
                
        return min(w1,w2)
              
        